File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        i,j=0,len(nums)-1
        mini = 1000
        while i<j:
            mid = (i+j)//2
            if nums[mid] > nums[j]:
                i = mid+1
            else:
                j = mid
        return nums[i]

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-3.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int> dp(nums.size(),1);

        for(int i=1; i<nums.size(); i++){
            for(int j=0; j<i; j++){
                if(nums[i] > nums[j]){
                    dp[i] = max(dp[i], 1+dp[j]);
                    cout<<dp[i];
                }
            }
        }
        int max_val = 1;
        for(int i=0; i<nums.size(); i++){
            max_val = max(max_val, dp[i]);
        }
        return max_val;
    }
};
